### PR TITLE
tiny terminal-util.c fixlets

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: src/basic/terminal-util.c [hunk 1]
```diff
@@ -898,11 +898,10 @@ int getttyname_malloc(int fd, char **ret) {
         assert(ret);
 
         r = ttyname_r(fd, path, sizeof path); /* positive error */
-        assert(r >= 0);
         if (r == ERANGE)
                 return -ENAMETOOLONG;
-        if (r > 0)
-                return -r;
+        if (r != 0)
+                return negative_errno();
```
Comment (keszybz) [inline src/basic/terminal-util.c]: Hmm, the man page says `ttyname_r` returns "error number upon error". I don't think we should use errno here.

Comment (poettering) [reply src/basic/terminal-util.c]: oops

**Comment by poettering** [comment]
force pushed a new version with the commit @keszybz indicated droppped

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**